rewrite-java/src/main/java/org/openrewrite/java/cleanup/RenameExceptionInEmptyCatch.java

```diff
@@ -0,0 +1,146 @@
+/*
+ * Copyright 2022 the original author or authors.
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * https://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.openrewrite.java.cleanup;
+
+import org.openrewrite.Cursor;
+import org.openrewrite.ExecutionContext;
+import org.openrewrite.Recipe;
+import org.openrewrite.TreeVisitor;
+import org.openrewrite.java.JavaIsoVisitor;
+import org.openrewrite.java.RenameVariable;
+import org.openrewrite.java.tree.J;
+
+import java.time.Duration;
+import java.util.*;
+
+public class RenameExceptionInEmptyCatch extends Recipe {
+
+    @Override
+    public String getDisplayName() {
+        return "Rename caught exceptions in empty catch blocks to `ignored`";
+    }
+
+    @Override
+    public String getDescription() {
+        return "Renames caught exceptions in empty catch blocks to `ignored`. `ignored` will be incremented by 1 if a namespace conflict exists. ";
+    }
+
+    @Override
+    public Duration getEstimatedEffortPerOccurrence() {
+        return Duration.ofMinutes(2);
+    }
+
+    @Override
+    protected TreeVisitor<?, ExecutionContext> getVisitor() {
+        return new JavaIsoVisitor<ExecutionContext>() {
+            @Override
+            public J.CompilationUnit visitCompilationUnit(J.CompilationUnit cu, ExecutionContext executionContext) {
+                Map<Cursor, Set<String>> variableScopes = new LinkedHashMap<>();
+                executionContext.putMessage("VARIABLES_KEY", variableScopes);
+
+                Map<Cursor, Set<J.VariableDeclarations.NamedVariable>> emptyTryVariables = new HashMap<>();
+                executionContext.putMessage("EMPTY_TRY_KEY", emptyTryVariables);
+
+                J.CompilationUnit cus = super.visitCompilationUnit(cu, executionContext);
+                aggregateNameScopes(variableScopes);
+
+                String baseName = "ignored";
```

Comment:
This was a little more complicated than I had anticipated.

One thing that could make this a bit easier would be to collect variables at each scope (like you are doing)..and then in the visitVariableDeclarations, when you encounter an empty catch, you could just walk up the parent scope (in that method), look for variables in that scope...and then rinse and repeat until you get to the CU. Then you could schedule the rename variable in there and not manage a collection of things that need to be modifed.

Reply:
That's a good point - thank you, I'll update the recipe tomorrow.